pkg/kubelet/dockertools/kube_docker_client.go

```diff
@@ -77,20 +79,24 @@ const (
 	// is made for defaultImagePullingStuckTimeout, the image pulling will be cancelled.
 	// Docker reports image progress for every 512kB block, so normally there shouldn't be too long interval
 	// between progress updates.
-	// TODO(random-liu): Make this configurable
 	defaultImagePullingStuckTimeout = 1 * time.Minute
 )
 
 // newKubeDockerClient creates an kubeDockerClient from an existing docker client. If requestTimeout is 0,
 // defaultTimeout will be applied.
-func newKubeDockerClient(dockerClient *dockerapi.Client, requestTimeout time.Duration) DockerInterface {
+func newKubeDockerClient(dockerClient *dockerapi.Client, requestTimeout, imagePullingStuckTimeout time.Duration) DockerInterface {
 	if requestTimeout == 0 {
 		requestTimeout = defaultTimeout
 	}
 
+	if imagePullingStuckTimeout == 0 {
```

Comment:
Sorry that my original code is not very good.
We should export the variables and set default value [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/v1alpha1/defaults.go#L172).